### PR TITLE
feat: add allbridge deposit hook

### PR DESCRIPTION
**Pull request by MuckT**
### Description

- Adds the AllBridge pool Abi.
- Adds a deposit shortcut for AllBridge pools.

### Testing Plan
- [x] Test with Valora locally
- [x] Add Unit tests
- [x] [Successful deposit](https://celoscan.io/tx/0xea006d392f3408dcf114852e7a1e55f01a2458682e2a27a2d76b7341d7fa8ada) with hooks cli: 

```
yarn triggerShortcut --networkId celo-mainnet --address <userAddress> --positionAddress 0xfb2c7c10e731ebe96dabdf4a96d656bfe8e2b5af --app allbridge --shortcut deposit --token.address 0x48065fbbe25f71c9282ddf5e1cd6d6a887483d5e --token.amount 1 --token.decimals 6 --mnemonic "<mnemonic>" --derivationPath "<derivationPath>"
```

### Related Issued

Part of [ACT-1270](https://linear.app/valora/issue/ACT-1270)

### CODE REVIEW

File: src/apps/allbridge/abis/pool.ts [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+  },
+  {
+    inputs: [{ internalType: 'uint256', name: 'amountLp', type: 'uint256' }],
+    name: 'withdraw',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  { stateMutability: 'payable', type: 'receive' },
+]
```
Comment (finnian0826) [inline src/apps/allbridge/abis/pool.ts:555]:
```suggestion
] as const
```
This is the fix Jean helped me with

File: src/apps/allbridge/shortcuts.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import hook from './shortcuts'
+import { NetworkId } from '../../types/networkId'
+
+describe('getShortcutDefinitions', () => {
+  it('should get the definitions successfully', async () => {
+    const shortcuts = await hook.getShortcutDefinitions(
+      NetworkId['arbitrum-one'],
+    )
+    expect(shortcuts.length).toBeGreaterThan(0)
```
Comment (finnian0826) [inline src/apps/allbridge/shortcuts.e2e.ts:9]: Should we check for the expected shortcut IDs here?

File: src/apps/allbridge/shortcuts.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          }),
+          positionAddress: ZodAddressLowerCased,
+        },
+
+        async onTrigger({ networkId, address, token, positionAddress }) {
+          const walletAddress = address as Address
+          const transactions: Transaction[] = []
+
+          // amount in smallest unit
+          const amountToSupply = parseUnits(token.amount, token.decimals)
```
Comment (finnian0826) [inline src/apps/allbridge/shortcuts.ts:41]: Does this make the user deposit their full amount of the deposit token?

Comment (MuckT) [reply src/apps/allbridge/shortcuts.ts:41]: In the CLI the amount deposited is specified by `--token.amount` flag. I am unsure of how this flag is used by Valora within the dapp shortcuts page.

File: src/apps/allbridge/shortcuts.e2e.ts (hunk 2 above)
Comment (MuckT) [reply src/apps/allbridge/shortcuts.e2e.ts:9]: Added in e9dba5f2d289a26da9f6f01d7f9b89d0c7f2707e!

File: src/apps/allbridge/shortcuts.ts (hunk 3 above)
Comment (jeanregisser) [reply src/apps/allbridge/shortcuts.ts:41]: Yes there's no UI yet, earn shortcuts will be the first to pass those arguments.

**Review by jeanregisser — APPROVED** [review]
Awesome! 🚀

**Pull request merged**